common/dal2/src/ops/io.rs

```diff
@@ -117,3 +125,110 @@ where F: FnMut(usize)
         r
     }
 }
+/// Create a buffered `SeekableReader`
+///
+/// ## Note
+///
+/// We will use 4MB buffer for now to avoid too much repeated requests.
+#[allow(dead_code)]
+pub async fn new_buffered_seekable_reader<'d, S>(
+    da: Arc<crate::DataAccessor<'d, S>>,
+    key: &str,
+) -> Result<BufReader<SeekableReader<'d, S>>>
+where
+    S: super::Read<S> + super::Stat<S>,
+{
+    let o = da.stat(key).await?;
+
+    // We will create a BufReader with a large buf so it only read big chunk from underlying storage.
+    Ok(BufReader::with_capacity(
+        4 * 1024 * 1024, // 4 MiB
+        SeekableReader::new(da, key, o.size),
+    ))
```

Comment:
I think it might be beneficial to allow the buffer size to be adjusted.

Reply:
Makes sense. As it's a simple change which doesn't affect the whole design, how about adding it while we really need this feature?

Reply:
ok